src/analyzer/protocol/mysql/mysql-analyzer.pac

```diff
@@ -22,12 +22,19 @@ refine flow MySQL_Flow += {
 		if ( ${msg.version} == 9 || ${msg.version == 10} )
 			connection()->zeek_analyzer()->AnalyzerConfirmation();
 
+		// If the client requested SSL and didn't provide credentials, switch to SSL
+		if ( ${msg.version} == 10 && ( ${msg.v10_response.cap_flags} & CLIENT_SSL ) && ${msg.v10_response.credentials}->empty() )
+			{
+			connection()->zeek_analyzer()->StartTLS();
```

Comment:
The MySQL header comes with a size and that's used on the top-level unit as &length.

https://github.com/zeek/zeek/blob/33244ae75f21ba01b0c18b2db74615aca822b314/src/analyzer/protocol/mysql/mysql-protocol.pac#L163-L171

So I thought the scenario you point out can't happen because `$input.length()` would not be 0 if the client actually had appended credentials (and the credentials[] array therefore filled) and only 0 if there weren't actual credentials. But I might be assuming things here?

Reply:
Resolving due to probably working out in practice.